README.md

```diff
@@ -1,72 +1,101 @@
 # Girder web component library
 
 ## For users
+
 ### Installation
-    npm install @girder/components
-    
-or
 
-    yarn add @girder/components
+```bash
+npm install @girder/components
+# or
+yarn add @girder/components
+```
 
 ### Use as a prebuilt UMD module
 
 It's not necessary to install Girder web components yourself, you can import the prebuilt bundle
 into your page by including the following tags:
 
-    <script src="https://unpkg.com/vue/dist/vue.min.js"></script>
-    <script src="https://unpkg.com/@girder/components"></script>
-    <link rel="stylesheet" href="https://unpkg.com/@girder/components/dist/girder.css">
+```html
+<script src="https://unpkg.com/vue/dist/vue.min.js"></script>
+<script src="https://unpkg.com/@girder/components"></script>
+<link rel="stylesheet" href="https://unpkg.com/@girder/components/dist/girder.css">
+```
 
 This will expose all the library's components under the global variable `girder`, e.g.
 `girder.components.Upload`.
 
 ### Import components into your project
 
-If you're building your own downstream application, you can include individual components from
-the library. Either import the full UMD module:
+#### Girder RestClient
+
+Many components in this library will require a `girder.RestClient` named `girderRest` through provide/inject.  
+The client can be provided through any common ancestor.  For example:
+
+```javascript
+import girder from '@girder/components';
+// apiRoot should point to your girder instance
+const restClient = new girder.RestClient({ apiRoot: 'http://localhost:8080/api/v1' });
+
+// Provide "girderRest" to the root
+new Vue({
+  provide: { girderRest },
+  ...
+}).$mount('#app');
+```
+
+#### Components
+
+If you're building your own downstream application, you can include individual components from the library. Because these are vuetify components, your consumer application is responsible for creating the `v-app` container.  [Read more](https://vuetifyjs.com/en/layout/pre-defined#all-about-app).  See `src/main.js` for a more comprehensive example.
```

Comment:
I did an experiment to see what the Upload component looks like without a `v-app` containing it, and it worked fine. In the interest of keeping Vuetify as an internal detail whenever possible, what does this component look like if it's not inside `v-app`?

Reply:
Yes, the rendering changes for this app without v-app.  I'm utilizing some of the light/dark featuers.

> Vuetify requires the use of the v-app component. It should wrap your entire application and is the center point for much of the framework functionality. If for whatever reason you cannot use this element, you can mimic it from attributes and classes. Set the data-app attribute to true on the highest element available (not including body), and the application application--{light|dark} classes.

As far as I can tell, the mimicking doesn't work.  Without v-app, it looks like some CSS classes don't get loaded, so even though the DOM looks the same, v-app isn't present to cause the necessary assets to load.

Light/dark still works, but `color="primary"` is no longer functional.  I can use custom CSS to fix the things that are wrong, but that feels like a code smell.

Reply:
I'm pretty confident about classes not getting loaded without v-app.  I can force webpack to load the missing assets by putting an empty v-app back in as a leaf node anywhere in the DOM with `display:none;` and the styles come back.

Reply:
Thanks for investigating, we shall keep this in the docs then.